cmd/did-method-cli/createdidcmd/createdid.go

```diff
@@ -0,0 +1,378 @@
+		" Alternatively, this can be set with the following environment variable: " + sidetreeWriteTokenEnvKey
+
+	publicKeyFileFlagName  = "publickey-file"
+	publicKeyFileEnvKey    = "DID_METHOD_CLI_PUBLICKEY_FILE"
+	publicKeyFileFlagUsage = "publickey file include public keys for Trustbloc DID " +
+		" Alternatively, this can be set with the following environment variable: " + publicKeyFileEnvKey
+
+	serviceFileFlagName = "service-file"
+	serviceFileEnvKey   = "DID_METHOD_CLI_SERVICE_FILE"
+	serviceFlagUsage    = "publickey file include services for Trustbloc DID " +
+		" Alternatively, this can be set with the following environment variable: " + serviceFileEnvKey
+
+	recoveryKeyFlagName  = "recoverykey"
+	recoveryKeyEnvKey    = "DID_METHOD_CLI_RECOVERYKEY"
+	recoveryKeyFlagUsage = "The public key PEM used for recovery of the document. Example: --recoverykey 'MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEXlp4fWF5rgLthKr20tsJ0tBIE6UmrGuAC8iVG/DaedkSt7HihCx/t2BGjooduaKwEIOmPjx2zBsbkbFrYhhnVw' " +
+		" Alternatively, this can be set with the following environment variable: " + recoveryKeyEnvKey
+
+	recoveryKeyFileFlagName  = "recoverykey-file"
+	recoveryKeyFileEnvKey    = "DID_METHOD_CLI_RECOVERYKEY_FILE"
+	recoveryKeyFileFlagUsage = "The file that contains the public key PEM used for recovery of the document. Example: --recoverykeyfile ./recovery_public.key " +
+		" Alternatively, this can be set with the following environment variable: " + recoveryKeyFileEnvKey
+
+	updateKeyFlagName  = "updatekey"
+	updateKeyEnvKey    = "DID_METHOD_CLI_UPDATEKEY"
+	updateKeyFlagUsage = "The public key PEM used for validating the signature of the next update of the document. Example: --updatekey 'MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEFMy2n9jYZChYSjdhK9vUWvPjz9tzBcEa13Ye33haxFsT//3kGxOQhI7yb3MJsDvwLtdfLL6txM3RdOrmLABBvw' " +
+		" Alternatively, this can be set with the following environment variable: " + updateKeyEnvKey
+
+	updateKeyFileFlagName  = "updatekey-file"
+	updateKeyFileEnvKey    = "DID_METHOD_CLI_UPDATEKEY_FILE"
+	updateKeyFileFlagUsage = "The file that contains the public key PEM used for validating the signature of the next update of the document. Example: --updatekeyfile ./update_public.key " +
+		" Alternatively, this can be set with the following environment variable: " + updateKeyFileEnvKey
+)
+
+type publicKey struct {
+	SignatureType string   `json:"signatureType,omitempty"`
+	Purposes      []string `json:"purposes,omitempty"`
+	JWKPath       string   `json:"jwkPath,omitempty"`
+}
+
+// GetCreateDIDCmd returns the Cobra create did command.
+func GetCreateDIDCmd() *cobra.Command {
+	createDIDCmd := createDIDCmd()
+
+	createFlags(createDIDCmd)
+
+	return createDIDCmd
+}
+
+func createDIDCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "create-did",
+		Short: "Create TrustBloc DID",
+		Long:  "Create TrustBloc DID",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			rootCAs, err := getRootCAs(cmd)
```

Comment:
This is a big block of code - consider moving it to a separate function.

Reply:
done https://github.com/trustbloc/trustbloc-did-method/compare/5d3e2f1b4d83907e304eba88714c1d648de0903b..8a890dc22810f7db8f7079b7cd8713e400022afc